oak_containers_orchestrator/src/container_runtime.rs

```diff
@@ -101,84 +80,47 @@ pub async fn run(
         base
     };
 
-    // mount the utility dir (includes the ipc socket) into the container
-    let container_dev_path = {
-        let mut base = container_rootfs_path.clone();
-        base.push(crate::UTIL_DIR);
-        base
-    };
-    rmount_dir(
-        std::path::Path::new("/").join(crate::UTIL_DIR),
-        container_dev_path,
-    )
-    .await?;
-
-    // mount host /dev into the container
-    let container_dev_path = {
-        let mut base = container_rootfs_path.clone();
-        base.push("dev");
-        base
-    };
-    rmount_dir(std::path::PathBuf::from("/dev"), container_dev_path).await?;
-
-    // mount host /sys into the container
-    let container_sys_path = {
-        let mut base = container_rootfs_path.clone();
-        base.push("sys");
-        base
-    };
-    rmount_dir(std::path::PathBuf::from("/sys"), container_sys_path).await?;
-
-    // mount host /proc into the container
-    let container_proc_path = {
-        let mut base = container_rootfs_path.clone();
-        base.push("proc");
-        base
-    };
-    rmount_dir(std::path::PathBuf::from("/proc"), container_proc_path).await?;
-
     log::debug!(
         "Startup command: {:?}",
         oci_filesystem_bundle_config.process.args
     );
+
+    // systemd-nspawn claims to support OCI bundles, but unfortunately it doesn't support bundles
+    // in the 1.0.2-dev format. See https://github.com/systemd/systemd/issues/22148
```

Comment:
This raises the question, why are we using the `1.0.2-dev` format? Does the dev suffix connotate a standard that's still under development?

Pre #4280 we used 0.2.0, which supports smaller set of features. 

Imo if we use a newer version, we should probably sure that our runtime supports the attributes that the newer versions support, such as hooks.

Reply:
Version "1.0.2-dev" is what is generated by `runc spec`. I am not sure whether the -dev extension means it is still under development.

Reply:
Hmmm, gotcha I suspected that may be the case. It seems we may be able to use https://github.com/opencontainers/image-tools. Looks it it'a OCI own tool for creating and validating runtime bundles, and it supports older versions too.

Reply:
Ugh, the situation is at best a mess.

I tried manually changing the version to 1.0.0, which is palatable to `systemd-nspawn`, but wasn't able to get it to start directly on my desktop as it failed to change the working directory. Which is weird.

So: on one hand, `systemd-nspawn` seems to be rather lightweight and integrates well into the systemd infrastructure.
On the other hand, I don't know what's going on with the OCI bundles there.

If we can get systemd-nspawn to ingest a OCI bundle, then my first preference would be to that. If we can't, then maybe indeed runc is a better option.

Reply:
I like that nspawn is part of systemd, but it per https://github.com/systemd/systemd/pull/27651 it's undertested. Given that (and lack of widespread adoption) I think runc is the better choice, even if we can get nspawn to work.

Reply:
(once we merge https://github.com/project-oak/oak/pull/4334 our container images should be OCI spec compliant)

Reply:
attempted to spawn the OCI bundles generated after merging https://github.com/project-oak/oak/pull/4334 with nspawn, since they're using `"ociVersion": "1.0.0"` which nspawn should support. But it seems to encounter a silent error and the container is not actually executed. Tried debugging for a bit but to no avail.